crates/processing_render/src/geometry/layout.rs

```diff
@@ -0,0 +1,114 @@
+use bevy::prelude::*;
+
+use super::BuiltinAttributes;
+use crate::error::{ProcessingError, Result};
+
+// bevy requires an attribute id for each unique vertex attribute. we don't really want to
+// expose this to users, so we hash the attribute name to generate a unique id. in theory
+// there could be collisions, but in practice this should be fine?
+// https://en.wikipedia.org/wiki/Fowler%E2%80%93Noll%E2%80%93Vo_hash_function#FNV-1a_hash
+pub const fn hash_attr_name(s: &str) -> u64 {
+    let bytes = s.as_bytes();
+    let mut hash = 0xcbf29ce484222325u64;
```

Comment:
I think this is fine given the wiki. though part of me wonders if those values should somehow be labeled as offset basis and prime in the code, too? non-blocking comment